### ISSUE TITLE
Missing library in recent VCS checkouts

### ISSUE THREAD

**Issue by pmattern** [nomacs/nomacs#47]
On Linux recent VCS checkouts can be compiled successfully but fail to start throwing a warning `error while loading shared libraries: libnomacsGui.so: cannot open shared object file: No such file or directory`.

Problem can be seen as of 1a5a0c5, the predecessors failed to build for different reasons.
Regarding 87a6168 and aeadc94 `make` failed throwing
```
/usr/bin/ld: CMakeFiles/nomacs.dir/3rdparty/libwebp/src/utils/thread.c.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/lib/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
CMakeFiles/nomacs.dir/build.make:1912: recipe for target 'nomacs' failed
make[2]: *** [nomacs] Error 1
CMakeFiles/Makefile2:102: recipe for target 'CMakeFiles/nomacs.dir/all' failed
make[1]: *** [CMakeFiles/nomacs.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```
while in becc101 it's already `cmake` to fail, corresponding error message is
```
CMake Error at cmake/UnixBuildTarget.cmake:86 (LIST):
  list sub-command REMOVE_ITEM requires list to be present.
Call Stack (most recent call first):
  CMakeLists.txt:149 (include)
```

All tests on up to date Arch Linux x86_64.

**Comment by diemmarkus** [comment]
hi,
thanks for the hint. could you check if webp is really turned off in your cmake? - we don't need webp anymore and stefan turned it off in 1a5a0c5.

**Comment by sfiel** [comment]
yes, webp is turned off ... the REMOVE_ITEM error was also removed last week when updating the cmake to our new structure. Please check out the newest version, I have updated the build system on linux last friday

**Comment by pmattern** [comment]
The failure to build due to REMOVE_ITEM ceased in aeadc94, the one about WebP in 1a5a0c5.
Both were mentioned only as these failures may eventually have masked when the actual problem of this issue started.

This issue's actual problem is the failure to launch after compiling successfully as depicted in the first comment's first paragraph.
It started with 1a5a0c5 and persists in 93bcf2a.

**Comment by sfiel** [comment]
I have just installed a archlinux on a virtual machine and i have no problem at all compiling and running the actual master branch on it ... have you already tried to delete the cmake cache?

**Comment by pmattern** [comment]
Apparently make target install didn't get adjusted to some code changes:

Running `nomacs` in-tree works here, too. Also, all needed libraries are in place
```
$ find . -iname '*libnomacs*' 
./libnomacsGui.so
./libnomacsLoader.so
./libnomacsCore.so
```
But after installing (`make install`) binary `nomacs` doesn't find the libraries any longer
```
$ ldd /usr/local/bin/nomacs | grep libnomacs 
        libnomacsGui.so => not found
        libnomacsLoader.so => not found
        libnomacsCore.so => not found
```
which indeed cannot be found where they are expected
```
# updatedb && locate libnomacs 
/usr/local/lib/libnomacsGui.so
```
(in-tree results skipped from this output).

**Comment by sfiel** [comment]
Thanks ... I forgot to change the install targets.

it is fixed now, altough you have to run ldconfig manually ... Haven't found a possibility to make cmake do this

**Comment by pmattern** [comment]
Problem is fixed as of 9e98278.

The need to run `ldconfig` depends, among other on, the selection of directories which are searched for shared libraries by default, AFAIK.
E. g. on Arch Linux it does not have to be run if switch `-DCMAKE_INSTALL_PREFIX=/usr` had been added to the cmake invocation of nomacs, as `/usr/lib` is considered anyway. If on the other hand cmake was run without any optional switch running `ldconfig` won't do as `/usr/local/lib` still isn't considered. Rather, the latter needs to be added to the ld.so configuration files before running `ldconfig`.
But I do not think this has any real relevance as nomacs will be installed by distribution packages most of the times and packagers should make sure that shared libraries are taken into account during installation of binary packages if needed.

Issue can be closed, IMO.

**Comment by sfiel** [comment]
thanks a lot
